contrib/shell/test.sh

```diff
@@ -1,55 +0,0 @@
-#!/usr/bin/env bash
-# SPDX-License-Identifier: Apache-2.0
-# Copyright Authors of Cilium
-
-readonly t_reset=$(tput sgr0)
-readonly red=$(tput bold; tput setaf 1)
-readonly green=$(tput bold; tput setaf 2)
-readonly yellow=$(tput bold; tput setaf 3)
-
-# Watch a file or directory for changes and trigger some action at that time.
-#
-# $1 = File to watch
-# $2+ = Command and arguments
-function watchdo
-{
-    local FILE=$1
-    shift
-
-    if [ ! -z "$TESTPKGS" ]; then
-        echo -e "${yellow}Using TESTPKGS=\"$TESTPKGS\" for run.${t_reset}"
-    fi
-    echo -e "${yellow}Running \"$@\" on changes to \"$FILE\" ...${t_reset}"
-    while inotifywait -q -r -e move $FILE; do
-        eval "$@";
-        if [ $? == 0 ] ; then
-            echo -e "${yellow}$@${t_reset}: ${green}✔${t_reset}"
-        else
-            echo -e "${yellow}$@${t_reset}: ${red}✘${t_reset}"
-        fi
-    done
-}
```

Comment:
I happen to find this helper super useful, but I already had it in my `.bashrc` and I was the one who added this into the tree to share the example with others :) I don't mind removing it if folks are discovering these tricks through other mechanisms.

One terminal: Run this command to retrigger tests whenever a file changes, then open your editor and make changes. Terminal reruns the test every time you save the file.

Reply:
[I had something similar based on inotifywait, too.]

I guess it's not a bad idea to keep these helpers somewhere for the reason you describe, but I wonder how we might make them more discoverable? Would it make sense to document them somewhere in the docs for contributing?

Reply:
We could. I think it's fair to say that this is out-of-scope of the actual goals of the project, but given the usefulness it's nice to have this example hosted somewhere. We could equally use a gist or such for this. So far I think the bar for `contrib/` directory has just been "does any developer find this useful?", and just put things there. From time to time we may share about tips/tricks naturally on Slack or during community meetings. If you think it's worth documenting in the contribution docs, we could also do that.

Reply:
seems like there is no strong objection to remove these files, I will merge this in and keep an eye if there is any users complaining about the removal.